MathLib/LinAlg/Eigen/EigenTools.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "EigenTools.h"
+
+#include <logog/include/logog.hpp>
+
+#include "EigenMatrix.h"
+#include "EigenVector.h"
+
+namespace MathLib
+{
+
+void applyKnownSolution(EigenMatrix &A_, EigenVector &b_, const std::vector<std::size_t> &vec_knownX_id,
+		const std::vector<double> &vec_knownX_x, double /*penalty_scaling*/)
+{
+    typedef EigenMatrix::RawMatrixType SpMat;
+    auto &A = A_.getRawMatrix();
+    auto &b = b_.getRawVector();
+    const size_t n_rows = A.rows();
```

Comment:
auto const or std::size_t

There are some more of size_t's below in the for-loops
A very safe way is to write
```c++
for (auto i = decltype(n_rows){0}; i < n_rows; ++i)
```
because there will be no conversion by definition. (but it's a little verbose)